### ISSUE TITLE
Feature request: support more properties on Events

### ISSUE THREAD

**Issue by soumya92** [mdirkse/i3ipc#3]
Currently Events only contain the "change" field, which is sometimes insufficient or inefficient to use.

For example, the window title change event already contains the new title of the current window, so it would make sense to expose that in the event directly. Otherwise we need extract the focussed window from the entire tree, which could be costly and also potentially add a race condition.

Or in the case of workspace change events, there is no way to get the previously selected workspace without keeping track of the current workspace by hand, which is error-prone and duplicate effort since the workspace change event already includes the "old" workspace.

It would be nice if the Event struct was expanded to add support for these other events (even as just a map[string]interface{}). Thoughts?

**Comment by mdirkse** [comment]
Agreed! I think the events should expose all the info that is sent along. Probably best just to create dedicated structs.
